grabbit/src/test/groovy/com/twcable/grabbit/client/GrabbitConfigurationSpec.groovy

```diff
@@ -24,6 +24,143 @@ import spock.lang.Unroll
+        GrabbitConfiguration.ConfigurationException exception = thrown()
+        exception.errors == [excludePaths: "relative paths provided cannot begin or end with /, ./ or \\"]
+
+        where:
+        excludePath << ["/a", "\\\\b" , "./c"]
+    }
+
+    def "generates correct exclusions regardless of trailing slash for path: #excludePath"() {
+        given:
+        def input = """
+        {
+            "serverUsername" : "admin",
+            "serverPassword" : "admin",
+            "serverHost" : "localhost",
+            "serverPort" : "4503",
+            "deltaContent" : false,
+            "pathConfigurations" :  [
+            {
+                "path" : "${path}",
+                "excludePaths" : ["d", "e/ab", "q/fg"],
+                "workflowConfigIds" : []
+            }
+          ]
+        }
+        """
+        def expectedOutput = ["/content/a/b/d", "/content/a/b/e/ab", "/content/a/b/q/fg"]
+
+        when:
+        def actualOutput = GrabbitConfiguration.create(input)
+
+        then:
+        actualOutput instanceof GrabbitConfiguration
+        actualOutput.pathConfigurations.first().excludePaths == expectedOutput
+
+        where:
+        path << ["/content/a/b", "/content/a/b/"]
+    }
+
+    @Unroll
+    def "Should fail to process exclude paths"() {
+        when:
+        GrabbitConfiguration.create(input)
+
+        then:
+        final GrabbitConfiguration.ConfigurationException exception = thrown()
+        exception.errors == errors
+
+        where:
+        input              | errors
+        """
+            {
+                "serverUsername" : "admin",
+                "serverPassword" : "admin",
+                "serverHost" : "localhost",
+                "serverPort" : "4503",
+                "deltaContent" : false,
+                "pathConfigurations" :  [
+                    {
+                        "path" : "/content/a/b",
+                        "excludePaths" :["b"],
+                        "workflowConfigIds" : []
+                    },
+                    {
+                        "path" : "/content/a/c",
+                        "excludePaths" :["b"],
+                        "workflowConfigIds" :
+                        [
+                            "/etc/workflow/launcher/config/update_asset_mod",
+                            "/etc/workflow/launcher/config/update_asset_create",
+                            "/etc/workflow/launcher/config/dam_xmp_nested_writeback",
+                            "/etc/workflow/launcher/config/dam_xmp_writeback"
+                        ]
+                    },
+                    {
+                        "path" : "/content/dam/d/images",
+                        "excludePaths" :["f",null],
+                        "workflowConfigIds" :
+                        [
+                            "something"
+                        ]
+                    }
+                ]
+            }
+            """ | [excludePaths: "contains null/empty"]
```

Comment:
When there's only one set of inputs, `where:` is adding complexity with no benefit.